File: php/src/Model/NotificationChannel.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A delivery target. The signing secret is write-only and never returned.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class NotificationChannel
{
    public function __construct(
        public readonly string $id,
        public readonly string $type,
        public readonly string $target,
        /** @var list<string> */
        public readonly array $events,
        public readonly bool $disabled,
        /** ISO-8601 */
        public readonly string $createdAt,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            type: (string) ($data['type'] ?? ''),
            target: (string) ($data['target'] ?? ''),
            events: \array_map('strval', $data['events'] ?? []),
            disabled: (bool) ($data['disabled'] ?? false),
            createdAt: (string) ($data['createdAt'] ?? ''),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/NotificationChannel.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * A delivery target. The signing secret is write-only and never returned.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class NotificationChannel {
    private String id;
    private String type;
    private String target;
    private List<String> events;
    private boolean disabled;
    private String createdAt;  // ISO-8601

    public String getId() { return id; }
    public String getType() { return type; }
    public String getTarget() { return target; }
    public List<String> getEvents() { return events; }
    public boolean isDisabled() { return disabled; }
    public String getCreatedAt() { return createdAt; }
}
